gems/aws-sdk-core/spec/aws/xml/doc_builder_spec.rb

```diff
@@ -5,22 +5,22 @@ module Aws
   module Xml
     describe DocBuilder do
 
-      let(:result) { '' }
+      let(:result) { ::String.new }
 
       let(:options) { { target: result, indent: '' } }
 
       let(:xml) { DocBuilder.new(options) }
 
       it 'creates empty xml documents' do
         xml.node('Xml')
-        expect(result).to eq('<Xml/>')
+        expect(xml.target).to eq('<Xml/>')
```

Comment:
Why were these changed?

Reply:
I tried many things around buffers.  
This is a failed experiment where I attempt to make the string usage of the class immutable.  

Nonetheless, if we use `DocBuilder.new(target: result)` interface, and test the `result` input, it is implied that we wholeheartedly accept mutable strings. That is why I leave the change.

The actions are on me I will investigate these whole buffer and doc builder mess and come up with a better solution/explanation.